lib/htmllint.js

```diff
@@ -23,18 +30,45 @@ module.exports = function(config, done) {
     return str;
   }
 
+  // copy environment but filter java options to pass on the command line
+  function cleanEnvironment() {
+    Object.keys(process.env)
+      .filter(function(property){
+        if(property === 'JAVA_TOOL_OPTIONS' || property === '_JAVA_OPTIONS'){
+          extraOpts.push(process.env[property]);
+          return false;
+        }
+        return true;
+      })
+      .forEach(function(property){
+        environment[property] = process.env[property];
```

Comment:
Might be easier to just do a `forEach` loop then push/assign in an if/else. Separate `filter()` and `forEach()` calls just make it more complicated.